File: shared/src/iosMain/kotlin/com/example/snaplearn/di/KoinHelper.kt
package com.example.snaplearn.di

import com.example.snaplearn.data.AIService
import com.example.snaplearn.data.HistoryRepository
import com.example.snaplearn.data.TextRecognizer
import com.example.snaplearn.data.TextRecognizerFactory
import com.example.snaplearn.viewmodel.SharedAnswerViewModel
import com.example.snaplearn.viewmodel.SharedHistoryViewModel
import org.koin.core.component.KoinComponent
import org.koin.core.component.get
import org.koin.core.context.startKoin
import org.koin.dsl.module

/**
 * Helper class to initialize Koin in iOS and provide access to dependencies.
 */
object KoinHelper {
    /**
     * Initialize Koin for iOS.
     */
    fun initKoin() {
        startKoin {
            modules(
                AppModule.commonModule,
                AppModule.platformModule(),
                module { }
            )
        }
    }
}

/**
 * Provides access to dependencies in iOS.
 */
class IosKoinHelper : KoinComponent {
    fun getTextRecognizer(): TextRecognizer {
        return get<TextRecognizerFactory>().createTextRecognizer()
    }
    
    fun getAIService(): AIService {
        return get()
    }
    
    fun getHistoryRepository(): HistoryRepository {
        return get()
    }
    
    fun getSharedAnswerViewModel(): SharedAnswerViewModel {
        return get()
    }
    
    fun getSharedHistoryViewModel(): SharedHistoryViewModel {
        return get()
    }
} 

File: iosApp/iosApp/KoinHelper.swift
import Foundation
import shared

class KoinHelper {
    private let koin = KoinKt.doInitKoin().koin
    
    func initKoin() {
        // This will trigger Koin initialization if not already done
        _ = KoinKt.doInitKoin()
    }
    
    // Get the TextRecognizer instance from Koin
    func getTextRecognizer() -> TextRecognizer {
        return koin.get(objCClass: TextRecognizer.self) as! TextRecognizer
    }
    
    // Get the AIService instance from Koin
    func getAIService() -> AIService {
        return koin.get(objCClass: AIService.self) as! AIService
    }
    
    // Get the HistoryRepository instance from Koin
    func getHistoryRepository() -> HistoryRepository {
        return koin.get(objCClass: HistoryRepository.self) as! HistoryRepository
    }
    
    // Get SharedAnswerViewModel from Koin
    func getAnswerViewModel() -> SharedAnswerViewModel {
        return koin.get(objCClass: SharedAnswerViewModel.self) as! SharedAnswerViewModel
    }
    
    // Get SharedHistoryViewModel from Koin
    func getHistoryViewModel() -> SharedHistoryViewModel {
        return koin.get(objCClass: SharedHistoryViewModel.self) as! SharedHistoryViewModel
    }
} 